### PR TITLE
⚠️ Resolved and Resources become optional pointer types

### PR DESCRIPTION
**Pull request by mdbooth**
This updates Resolved and Resources in both OpenStackMachine and the Bastion to be optional pointer types.

TODO:
- [x] Merge https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1965

/hold

### CODE REVIEW

File: api/v1alpha7/openstackcluster_conversion.go [hunk 1]
```diff
@@ -348,14 +348,11 @@ func restorev1alpha7ClusterStatus(previous *OpenStackClusterStatus, dst *OpenSta
 }
 
 func restorev1beta1ClusterStatus(previous *infrav1.OpenStackClusterStatus, dst *infrav1.OpenStackClusterStatus) {
-	// ReferencedResources have no equivalent in v1alpha7
-	if previous.Bastion != nil {
-		dst.Bastion.Resolved = previous.Bastion.Resolved
+	if previous == nil || dst == nil {
```
Comment (MaysaMacedo) [inline api/v1alpha7/openstackcluster_conversion.go:351]: this check also happens in `restorev1beta1BastionStatus`, can it be removed?

**Review by MaysaMacedo — COMMENTED** [review]
Other than the nit I noticed, it LGTM.

File: api/v1alpha7/openstackcluster_conversion.go (hunk 1 above)
Comment (dulek) [reply api/v1alpha7/openstackcluster_conversion.go:351]: In `restorev1beta1BastionStatus` it's BastionStatus, not ClusterStatus. I think we need both.

Comment (lentzi90) [reply api/v1alpha7/openstackcluster_conversion.go:351]: Here we are checking `previous` and `dst`. Then we pass on `previous.Bastion` and `dst.Bastion`, so I would say both checks are needed.

**Review by lentzi90 — COMMENTED** [review]
/lgtm

**Pull request merged**